File: common/src/main/java/com/goexp/common/db/mongo/AbstractDBTemplate.java
package com.goexp.common.db.mongo;

import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoClients;

public abstract class AbstractDBTemplate {

    protected static MongoClient mongoClient;

    static {
        mongoClient = MongoClients.create();
    }

    protected final String dbName;
    protected final String tableName;

    public AbstractDBTemplate(String dbName, String tableName) {
        this.dbName = dbName;
        this.tableName = tableName;
    }
}

File: db_mongo/src/main/scala/com/goexp/common/db/mongo/AbstractDBTemplate.scala
package com.goexp.common.db.mongo

import java.util.Objects

import com.mongodb.client.{MongoClient, MongoClients}

object AbstractDBTemplate {
  private val mongoClient: MongoClient = MongoClients.create
}

abstract class AbstractDBTemplate(protected val dbName: String,
                                  protected val tableName: String) {
  protected val mongoClient = AbstractDBTemplate.mongoClient

  Objects.requireNonNull(dbName)
  Objects.requireNonNull(tableName)


}